### PR TITLE
feat(challenge-editor): make work in gitpod

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes issue where Naomi does not want contributors to have an easy time using Gitpod.

### CODE REVIEW

**Review by majestic-owl448 — DISMISSED** [review]
Oh, it works! Finally! 

LGTM :+1:

**Review by Sboonny — DISMISSED** [review]
Thank you, I always thought I was doing something wrong.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
So, I think we should these environment variables similarly to the rest.  Namely, have a .env file and import them from that, rather than falling back to hard-coded values if the Gitpod specific stuff is not set.

That would mean that rather than passing in `GITPOD_CLIENT_LOCATION` and `REACT_APP_GITPOD_API_LOCATION` directly, we can just update the .bashrc to override them.


Finally, since they're not really anything to do with Gitpod we can call them `CHALLENGE_EDITOR_CLIENT_LOCATION` and `REACT_APP_CHALLENGE_EDITOR_API_LOCATION`.

**Comment by ShaunSHamilton** [comment]
@ojeytonwilliams I did not want to clog up the `.env` with this stuff.

The reason I call it `GITPOD_` is because it **is** everything to do with Gitpod. I create the variable `API_LOCATION`, because it is one of Gitpod or other, but the Gitpod env variables are only set by Gitpod.

I suggest we still go with this, and we can discuss a different solution another time.

**Comment by ojeytonwilliams** [comment]
> I did not want to clog up the .env with this stuff.

Yeah, I understand.  The challenge editor is a tool that just happens to reside in the fcc repo, so I think it should have it's own environment variables.

> The reason I call it GITPOD_ is because it is everything to do with Gitpod. I create the variable API_LOCATION, because it is one of Gitpod or other, but the Gitpod env variables are only set by Gitpod.

They're set by Gitpod, sure, but that's the extent of Gitpod's involvement.  It's analogous to `HOME_LOCATION` - we use 

```
export HOME_LOCATION=$(gp url 8000)
```

in the .gitpod.yml, rather than creating a new variable `GITPOD_HOME_LOCATION`.

TL;DR: all Gitpod config should be in the .gitpod.yml file

**Review by raisedadead — DISMISSED** [review]

**Review by ojeytonwilliams — DISMISSED** [review]
Assuming everything still functions without the extra 'before's, I'm happy now.

**Review by ojeytonwilliams — APPROVED** [review]
It's odd that .bashrc is not sourced for the client and server, but it appears not to be.

**Review by Sembauke — APPROVED** [review]
